File: lab0/LabMath.java
import static java.lang.Math.*;

public class LabMath{
    public static void main(String[] args){
        double min_b = -12.0, max_b = 2.0;

        short[] a = new short[10];
        float[] x = new float[10];

        for(short i = 5, j = 0; i <= 23; i+=2, j++){
            a[j] = i;
            x[j] = (float) getRandom(min_b, max_b);
        }

        float[][] new_a = new float[10][10];
        for(int i = 0; i < 10; i++){
            for(int j = 0; j < 10; j++){
                switch (a[i]) {
                    case 13:
                        float c = (float) atan(pow((x[j] - 5) / 14, 2)) - 1;
                        float b = (float) pow(atan((x[j] - 5) / 14), 2 * (1 - exp(x[j])));
                        new_a[i][j] = (float) pow(sin(cbrt(x[j])), c/b);
                        break;
                    case 5:
                    case 11:
                    case 15:
                    case 19:
                    case 21:
                        new_a[i][j] = (float) pow(2 * cbrt(cos(x[j])), 2.0);
                        break;
                    default:
                        new_a[i][j] = (float) cos(cos(cos(pow((3 - x[j]) / 2, x[j]))));
                        break;
                }
                System.out.printf("%.2f\t", new_a[i][j]);
            }
            System.out.println();
        }
        
    }
    private static double getRandom(double min, double max){
        return Math.random()*((max-min)+1.0)+min;
    }

}

File: lab0/LabMath.kt
import java.lang.Math.*

fun main(){
    val min_b: Double = -12.0
    val max_b: Double = 2.0

    var a = ShortArray(10)
    var x = FloatArray(10)
    
    var j: Int = 0
    for(i in 5..23 step 2){
        a[j] = i.toShort()
        x[j] = getRandom(min_b, max_b).toFloat()
        j++
    }

    var new_a = Array(10){FloatArray(10)}
    for(i in 0..9){
        for(j in 0..9){
            when(a[i].toInt()){
                13 -> {
                    var c = (atan(pow((x[j].toDouble() - 5.0) / 14.0, 2.0)) - 1)
                    var b = (pow(atan((x[j].toDouble() - 5.0) / 14.0), 2.0 * (1 - exp(x[j].toDouble()))))
                    new_a[i][j] = pow(sin(cbrt(x[j].toDouble())), c / b).toFloat()
                }
                5, 11, 15, 19, 21 -> {
                    new_a[i][j] = pow(2 * cbrt(cos(x[j].toDouble())), 2.0).toFloat()
                }
                else -> {
                    new_a[i][j] = cos(cos(cos(pow((3 - x[j].toDouble()) / 2, x[j].toDouble())))).toFloat()
                }
            }
            print("%.2f \t".format(new_a[i][j]))
        }
        println()
    }
    
}

fun getRandom(min: Double, max:Double): Double{
    return random()*(max - min + 1.0) + min
}